File: RecyclerCardView/app/src/main/java/com/nayla/recyclercardview/SiswaAdapter.java
package com.nayla.recyclercardview;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class SiswaAdapter extends RecyclerView.Adapter<SiswaAdapter.Viewolder> {

    private Context context;
    private List<Siswa>siswaList;

    public SiswaAdapter(Context context, List<Siswa> siswaList) {
        this.context = context;
        this.siswaList = siswaList;
    }

    @NonNull
    @Override
    public Viewolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_siswa, viewGroup, false);
        return new Viewolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull Viewolder viewHolder, int i) {
        Siswa siswa = siswaList.get(i);
        viewHolder.tvNama.setText(siswa.getNama());
        viewHolder.tvAlamat.setText(siswa.getAlamat());

        viewHolder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(context, "Nama :"+siswa.getNama()+"Alamat : "+siswa.getAlamat(), Toast.LENGTH_SHORT).show();
            }
        });
    }



    @Override
    public int getItemCount() {
        return siswaList.size();
    }

    public class Viewolder extends RecyclerView.ViewHolder{

        TextView tvNama, tvAlamat;


        public Viewolder(@NonNull View itemView) {
            super(itemView);

            tvNama = itemView.findViewById(R.id.tvNama);
            tvAlamat = itemView.findViewById(R.id.tvAlamat);
        }
    }
}

File: LaravelApi/app/src/main/java/com/jihannayla/laravelapi/adapter/SiswaAdapter.kt
package com.jihannayla.laravelapi.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.jihannayla.laravelapi.databinding.ListSiswaAdapterBinding
import com.jihannayla.laravelapi.model.DataSiswa

class SiswaAdapter (
    private val listSiswa: List<DataSiswa?>?,
): RecyclerView.Adapter<SiswaAdapter.ViewHolder>() {
    inner class ViewHolder(val ListSiswaAdapterBinding: ListSiswaAdapterBinding):
    RecyclerView.ViewHolder(ListSiswaAdapterBinding.root) {
        fun onBindItem(dataSiswa: DataSiswa?) {
            ListSiswaAdapterBinding.namasiswa.text = dataSiswa?.namasiswa
            ListSiswaAdapterBinding.nisn.text = dataSiswa?.nisn
            ListSiswaAdapterBinding.alamat.text = dataSiswa?.alamat
            ListSiswaAdapterBinding.agama.text = dataSiswa?.agama
            ListSiswaAdapterBinding.usia.text = dataSiswa?.usia
            ListSiswaAdapterBinding.jeniskelamin.text = dataSiswa?.gender

        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SiswaAdapter.ViewHolder {
        val binding =
            ListSiswaAdapterBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return  ViewHolder(binding)

    }

    override fun onBindViewHolder(holder: SiswaAdapter.ViewHolder, position: Int) {
        holder.onBindItem(listSiswa?.get(position))

    }

    override fun getItemCount(): Int {
        return  listSiswa?.size?:0
    }

}